File: Runtime/StratumCollection.vb
'*********************************************************************************************
' ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

Imports System.Collections.ObjectModel

''' <summary>
''' The stratum collection class
''' </summary>
''' <remarks></remarks>
Friend Class StratumCollection
    Inherits KeyedCollection(Of Integer, Stratum)

    Protected Overrides Function GetKeyForItem(ByVal item As Stratum) As Integer
        Return item.StratumId
    End Function

End Class

File: src/Runtime/StratumCollection.cs
﻿// A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System.Collections.ObjectModel;

namespace SyncroSim.STSim
{
    internal class StratumCollection : KeyedCollection<int, Stratum>
    {
        protected override int GetKeyForItem(Stratum item)
        {
            return item.StratumId;
        }
    }
}